src/components/Home/SeeThroughLens.tsx

```diff
@@ -0,0 +1,143 @@
+import UserProfile from '@components/Shared/UserProfile';
+import { Input } from '@components/UI/Input';
+import { Spinner } from '@components/UI/Spinner';
+import type { Profile } from '@generated/types';
+import { CustomFiltersTypes, SearchRequestTypes, useSearchProfilesLazyQuery } from '@generated/types';
+import { Menu, Transition } from '@headlessui/react';
+import { XIcon } from '@heroicons/react/outline';
+import { ChevronDownIcon } from '@heroicons/react/solid';
+import getAvatar from '@lib/getAvatar';
+import { Leafwatch } from '@lib/leafwatch';
+import clsx from 'clsx';
+import type { ChangeEvent } from 'react';
+import React, { Fragment, useState } from 'react';
+import { useAppStore } from 'src/store/app';
+import { useTimelineStore } from 'src/store/timeline';
+import { SEARCH } from 'src/tracking';
+
+const SeeThroughLens = () => {
+  const currentProfile = useAppStore((state) => state.currentProfile);
+  const seeThroughProfile = useTimelineStore((state) => state.seeThroughProfile);
+  const setSeeThroughProfile = useTimelineStore((state) => state.setSeeThroughProfile);
+  const recommendedProfilesToSeeThrough = useTimelineStore((state) => state.recommendedProfilesToSeeThrough);
+  const [searchText, setSearchText] = useState('');
+
+  const profile = seeThroughProfile ?? currentProfile;
+
+  const [searchUsers, { data: searchUsersData, loading: searchUsersLoading }] = useSearchProfilesLazyQuery();
+
+  const handleSearch = (evt: ChangeEvent<HTMLInputElement>) => {
+    const keyword = evt.target.value;
+    setSearchText(keyword);
+    searchUsers({
+      variables: {
+        request: {
+          type: SearchRequestTypes.Profile,
+          query: keyword,
+          customFilters: [CustomFiltersTypes.Gardeners],
+          limit: 5
+        }
+      }
+    });
+  };
+
+  // @ts-ignore
+  const searchResults = searchUsersData?.search?.items ?? [];
+  const recommendedProfiles = recommendedProfilesToSeeThrough ?? [];
+
+  const profiles =
+    searchResults.length && searchText.length ? searchResults : recommendedProfiles.slice(0, 5);
+
+  return (
+    <Menu as="div" className="relative">
+      <Menu.Button className="rounded-md hover:bg-gray-300 p-1 hover:bg-opacity-20">
+        <span className="flex space-x-1 items-center text-sm">
+          <img
+            src={getAvatar(profile)}
+            loading="lazy"
+            className="bg-gray-200 w-5 h-5 rounded-full border dark:border-gray-700/80"
+            alt={profile?.handle}
+          />
+          <span>{seeThroughProfile ? `@${profile?.handle}` : 'My Feed'}</span>
```

Comment:
@bigint should we consider having a component instead of always prefixing it with `@`?  WDYT